packages/angular-query-experimental/src/__tests__/query-options.test-d.ts

```diff
@@ -71,6 +71,16 @@ test('should tag the queryKey with unknown if there is no queryFn', () => {
   assertType<unknown>(queryKey[dataTagSymbol])
 })
 
+test('should tag the queryKey with the result type of the QueryFn if select is used', () => {
+  const { queryKey } = queryOptions({
+    queryKey: ['key'],
+    queryFn: () => Promise.resolve(5),
+    select: (data) => data.toString(),
+  })
+
+  assertType<unknown>(queryKey[dataTagSymbol])
```

Comment:
wondering why this would be `unknown`. Shouldn't it be `string` ?

```suggestion
  assertType<string>(queryKey[dataTagSymbol])
```